src/bsa/plugin.cpp

```diff
@@ -187,13 +197,22 @@ auto list_archive(const Path &dir, const Settings &sets) noexcept -> std::vector
 {
     try
     {
-        return flux::from_range(fs::directory_iterator(dir))
+        std::vector<Path> archives = flux::from_range(fs::directory_iterator(dir))
             .filter([](const auto &f) { return f.is_regular_file(); })
             .filter([&sets](const auto &f) {
                 return common::to_lower(f.path().extension().u8string()) == sets.extension;
             })
             .map([](const auto &f) { return f.path(); })
             .to<std::vector>();
+
+        // When CAO splits archives, it adds a suffix to the path. This mean we can easily
+        // sort the paths to get the first archive in the first spot to name the dummy plugin
+        // correctly (in case only one is created).
+        std::sort(archives.begin(), archives.end(), [](const auto p1, const auto p2) {
+            return p1.stem() <= p2.stem();
+        });
```

Comment:
You can use [flux::sort](https://tristanbrindle.com/flux/reference/algorithms.html#_CPPv4I_22random_access_sequence0EIQ9see_belowEN4flux4sortEvRR3Seq3Cmp) and chain it to the previous lines

Reply:
I tried, but I couldn't get it to compile. I will be completely honest in saying that I do not understand `flux` technically, the documentation is super sparse.

Reply:
Don't worry about that, this library is very new and not really ready for production, but I just love it